### PR TITLE
feat(rancher): open modal in url path

### PR DESCRIPTION
**Pull request by lionel95200x**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-14086
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Open modal using custom URL 

- Edit Name /rancher/:RancherId/edit-name
- Generate access   /rancher/:RancherId/generate-access
- Delete Rancher /rancher/:RancherId/delete

Retest all modal behavior (Edit name, Generate Access, Delete Rancher)

### CODE REVIEW

**Review by aboungnaseng-ovhcloud — DISMISSED** [review]

File: packages/manager/apps/pci-rancher/src/components/Table/TableContainer.tsx [hunk 1]
```diff
@@ -8,53 +8,41 @@ import {
...
 export default function TableContainer({
   data,
-  refetchRanchers,
 }: Readonly<RancherDatagridWrapper>) {
   const { t } = useTranslation('pci-rancher/listing');
   const [sorting, setSorting] = useState<SortingState>([]);
-  const { projectId } = useParams();
+  const state = useMutationState({
+    filters: {
+      mutationKey: deleteRancherServiceQueryKey('').slice(0, 1),
```
Comment (frenautvh) [inline packages/manager/apps/pci-rancher/src/components/Table/TableContainer.tsx:33]:
```suggestion
      mutationKey: deleteRancherServiceQueryKey('')[0],
```

File: packages/manager/apps/pci-rancher/src/components/Table/TableContainer.tsx [hunk 2]
```diff
@@ -8,53 +8,41 @@ import {
...
-    onError: () => setDeleteRancherResponse(ODS_MESSAGE_TYPE.error),
-  });
 
   const Actions = ({ row }: { row: Row<RancherService> }) => (
     <ActionsCell
-      openModal={() => setIsShowDeleteModal(true)}
-      setSelectedRancher={setSelectedRancher}
       row={row}
+      onClickDelete={() =>
+        navigate(`${location.pathname}/${row.original.id}/delete`)
```
Comment (frenautvh) [inline packages/manager/apps/pci-rancher/src/components/Table/TableContainer.tsx:44]: couldn't you use relative path such as navigate('..') for example ?

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -30,41 +30,38 @@ export type DashboardTabItemProps = {
...
-  const { mutate: editRancherName } = useEditRancherName({
-    projectId: projectId as string,
-    rancherId: rancher.id,
-    onSuccess: () => {
-      setEditNameResponseType(ODS_MESSAGE_TYPE.success);
-      refetchRancher();
-    },
-    onError: () => setEditNameResponseType(ODS_MESSAGE_TYPE.error),
+  const mutationEditNameState = useMutationState({
+    filters: { mutationKey: patchRancherServiceQueryKey('').slice(0, 1) },
```
Comment (frenautvh) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx:42]:
```suggestion
    filters: { mutationKey: patchRancherServiceQueryKey('')[0] },
```

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -30,41 +30,38 @@ export type DashboardTabItemProps = {
...
-  const {
-    generateAccesDetail,
-    accessDetail,
-    hasErrorAccessDetail,
-  } = useGenerateAccessDetail({
-    projectId: projectId as string,
-    rancherId: rancher.id,
+  const mutationGenerateAccessState = useMutationState({
+    filters: {
+      mutationKey: postRancherServiceQueryKey('').slice(0, 1),
```
Comment (frenautvh) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx:47]:
```suggestion
      mutationKey: postRancherServiceQueryKey('')[0],
```

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail.tsx [hunk 5]
```diff
@@ -17,57 +17,43 @@ import {
...
-import EditNameModal from '../../Modal/EditNameModal';
+import { useHref } from 'react-router-dom';
 
 import { RancherService } from '@/api/api.type';
-import { TileBlock } from '../../TileBlock/TileBlock';
-import GenerateAccessModal from '../../Modal/GenerateAccesModal';
-import { AccessDetail } from '../../../hooks/useGenerateAccessDetail';
 import { useTrackingAction } from '@/hooks/useTrackingPage';
 import { TrackingEvent, TrackingPageView } from '@/utils/tracking';
+import { TileBlock } from '../../TileBlock/TileBlock';
```
Comment (frenautvh) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail.tsx:31]:
```suggestion
import { TileBlock } from '@/src/components/TileBlock/TileBlock';
```

File: packages/manager/apps/pci-rancher/src/pages/dashboard/edit-modal/EditModal.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import { useRancher } from '@/hooks/useRancher';
+import useEditRancherName from '@/hooks/useEditRancherName';
+
+export default function EditModal() {
+  const { data, refetch } = useRancher();
+  const { projectId } = useParams();
+
+  const navigate = useNavigate();
+
+  const rancher = data.data;
```
Comment (frenautvh) [inline packages/manager/apps/pci-rancher/src/pages/dashboard/edit-modal/EditModal.tsx:14]: does the fetch returns { data } already ? it looks weird having data.data

**Review by frenautvh — APPROVED** [review]

**Review by aboungnaseng-ovhcloud — DISMISSED** [review]

**Review by frenautvh — APPROVED** [review]

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

File: packages/manager/apps/pci-rancher/src/hooks/useRancher.tsx [hunk 7]
```diff
@@ -13,11 +20,43 @@ export const useRancher = () => {
 
 export const ranchersQueryKey = (projectId: string) => ['project', projectId];
 
-export const useRanchers = () => {
+export const useRanchers = ({
+  disablePooling,
```
Comment (anooparveti) [inline packages/manager/apps/pci-rancher/src/hooks/useRancher.tsx:24]:
```suggestion
  disablePolling,
```

File: packages/manager/apps/pci-rancher/src/pages/create/index.tsx [hunk 8]
```diff
@@ -36,7 +36,7 @@ export default function Create() {
 
   const { data: project } = usePciProject();
 
-  const { data: ranchers } = useRanchers();
+  const { data: ranchers } = useRanchers({ disablePooling: true });
```
Comment (anooparveti) [inline packages/manager/apps/pci-rancher/src/pages/create/index.tsx:39]:
```suggestion
  const { data: ranchers } = useRanchers({ disablePolling: true });
```

**Review by anooparveti — APPROVED** [review]

**Review by frenautvh — APPROVED** [review]

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Pull request merged**